aws-lc-rs/src/aead/aes_gcm.rs

```diff
@@ -10,45 +11,89 @@ use crate::error::Unspecified;
 use aws_lc::EVP_AEAD_CTX_seal_scatter;
 use std::ptr::null;
 
+use super::{NONCE_LEN, TAG_LEN};
+
 #[inline]
 #[allow(clippy::needless_pass_by_value)]
 pub(crate) fn aead_seal_separate(
     key: &AeadCtx,
     nonce: Nonce,
     aad: Aad<&[u8]>,
     in_out: &mut [u8],
-) -> Result<Tag, Unspecified> {
-    unsafe {
-        let aead_ctx = match key {
-            AeadCtx::CHACHA20_POLY1305(aead_ctx)
-            | AeadCtx::AES_128_GCM(aead_ctx)
-            | AeadCtx::AES_256_GCM(aead_ctx) => aead_ctx,
-        };
-
-        let aad_slice = aad.as_ref();
+) -> Result<(Nonce, Tag), Unspecified> {
+    let aead_ctx = key.as_ptr();
+
+    let aad_slice = aad.as_ref();
+    let mut tag = [0u8; TAG_LEN];
```

Comment:
* Should we be consistent in using `MAX_TAG_LEN` for the size of such arrays? 
* `MAX_TAG_LEN` and `TAG_LEN` are currently the same value (and might always be). `MAX_TAG_LEN` is the value we expose publicly -- should we get rid of `TAG_LEN` until we have the need for some tag length constant that's a different value.

Reply:
I took a stab at doing a broader refactor to try to abstract out the hard-coding of TAG_LEN / NONCE_LEN where possible, and utilize the `tag_len()` and `nonce_len()` from the configured algorithm. That will be in the next revision I push.